vllm_spyre/v1/worker/spyre_model_runner.py

```diff
@@ -2302,6 +2302,14 @@ def _maybe_prepare_last_prefill(self, req_id: str,
         if num_computed_tokens + num_scheduled_tokens < prompt_len:
             return
 
+        # Last prefill: we might need to update the tkv
+        req_n_blocks = len(self._get_blocks(req_id))
+        max_n_blocks = max(req_n_blocks,
+                           max(0, self.tkv - 1) // self.block_size)
+        n_left_padding = (max_n_blocks - req_n_blocks) * self.block_size
```

Comment:
`self._get_blocks(req_id)` doesn't include padding blocks, so this expression won't correctly result in the number of padding blocks, I think.